src/dec128.rs

```diff
@@ -436,6 +436,15 @@ impl d128 {
             }
         }
     }
+    
+    /// Hidden function for initializing a d128 from an array of bytes. When using this function
+    /// you should be careful about endianness issues.
+    #[doc(hidden)]
+    pub fn from_bytes(bytes: [u8; 16]) -> d128 {
```

Comment:
Unfortunately this is the thing that I struggled with. If we use `dec128::from_hex()` won't that just negate the performance benefits, and being able to use `d128` in constant expressions?

The major problem is that it can't store a resolved identifier, because the compiler hasn't done that stage yet (I think). There was an issue somewhere on it, I should've noted it down.